File: PY/tendency.py
import pandas as pd
import matplotlib.pyplot as plt
import statistics
import numpy as np
from scipy.stats import linregress

# Leer los datos del archivo resultados.csv
data = pd.read_csv("resultados.csv")

velocidad_subida = data["up"]
velocidad_bajada = data["down"]

# Cálculos estadísticos de subida
media_subida = statistics.mean(velocidad_subida)
moda_subida = statistics.mode(velocidad_subida)
desviacion_subida = statistics.stdev(velocidad_subida)

# Cálculos estadísticos de bajada
media_bajada = statistics.mean(velocidad_bajada)
moda_bajada = statistics.mode(velocidad_bajada)
desviacion_bajada = statistics.stdev(velocidad_bajada)

# Gráfica de dispersión de velocidad de subida y bajada
plt.figure(figsize=(8, 6))
plt.scatter(velocidad_subida, velocidad_bajada)
plt.xlabel("Velocidad de subida (Mbps)")
plt.ylabel("Velocidad de bajada (Mbps)")
plt.title("Velocidad de Subida vs Velocidad de Bajada")

# Perform linear regression
slope, intercept, r_value, p_value, std_err = linregress(
    velocidad_subida, velocidad_bajada
)

# Generate points for the trend line
x = np.linspace(min(velocidad_subida), max(velocidad_subida), 100)
y = slope * x + intercept

# Plot the trend line
plt.plot(x, y, color="red")

# Print the trendline equation and r-value
equation = f"y = {slope:.2f}x + {intercept:.2f}"
r_squared = f"r = {r_value:.2f}"

plt.text(40, 260, equation, fontsize=12)
plt.text(40, 250, r_squared, fontsize=12)

# Resultados de subida
print("Estadísticas de Velocidad de Subida:")
print("Media:", media_subida)
print("Moda:", moda_subida)
print("Desviación estándar:", desviacion_subida)

# Resultados de bajada
print("\nEstadísticas de Velocidad de Bajada:")
print("Media:", media_bajada)
print("Moda:", moda_bajada)
print("Desviación estándar:", desviacion_bajada)

print()
print(equation)
print(r_squared)

plt.show()

File: tendency.R
library(ggplot2)
library(stats)

# Leer los datos del archivo resultados.csv
data <- read.csv("results.csv")

velocidad_subida <- data$up
velocidad_bajada <- data$down

# Cálculos estadísticos de subida
media_subida <- mean(velocidad_subida)
moda_subida <- as.numeric(names(table(velocidad_subida))[table(velocidad_subida) == max(table(velocidad_subida))])
desviacion_subida <- sd(velocidad_subida)

# Cálculos estadísticos de bajada
media_bajada <- mean(velocidad_bajada)
moda_bajada <- as.numeric(names(table(velocidad_bajada))[table(velocidad_bajada) == max(table(velocidad_bajada))])
desviacion_bajada <- sd(velocidad_bajada)

# Gráfica de dispersión de velocidad de subida y bajada
ggplot(data, aes(x = up, y = down)) +
  geom_point() +
  labs(x = "Velocidad de subida (Mbps)", y = "Velocidad de bajada (Mbps)",
       title = "Velocidad de Subida vs Velocidad de Bajada") +
  geom_smooth(method = "lm", se = FALSE, color = "red") +
  geom_text(x = 40, y = 260, label = paste("y =", round(coef(lm(down ~ up, data))["up"], 2), "x +",
                                           round(coef(lm(down ~ up, data))["(Intercept)"], 2)), size = 4) +
  geom_text(x = 40, y = 250, label = paste("r =", round(cor(velocidad_subida, velocidad_bajada), 2)), size = 4)

# Resultados de subida
cat("Estadísticas de Velocidad de Subida:\n")
cat("Media:", media_subida, "\n")
cat("Moda:", moda_subida, "\n")
cat("Desviación estándar:", desviacion_subida, "\n")

# Resultados de bajada
cat("\nEstadísticas de Velocidad de Bajada:\n")
cat("Media:", media_bajada, "\n")
cat("Moda:", moda_bajada, "\n")
cat("Desviación estándar:", desviacion_bajada, "\n")

# Perform linear regression
lm_model <- lm(down ~ up, data)
slope <- coef(lm_model)["up"]
intercept <- coef(lm_model)["(Intercept)"]
r_value <- cor(velocidad_subida, velocidad_bajada)

# Print the trendline equation and r-value
cat("\n")
cat("y =", round(slope, 2), "x +", round(intercept, 2), "\n")
cat("r =", round(r_value, 2), "\n")
